internal/protocol/tcp_server.go

```diff
@@ -30,9 +33,18 @@ func TCPServer(listener net.Listener, handler TCPHandler, logf lg.AppLogFunc) er
 			}
 			break
 		}
-		go handler.Handle(clientConn)
+
+		wg.Add(1)
+
+		go func() {
+			defer wg.Done()
```

Comment:
minor nit: you do not need `defer` because there are not multiple return paths from this function. So it's a tiny bit simpler and a tiny bit more efficient to just put this statement second, without `defer`.